src/Illuminate/Collections/Collection.php

```diff
@@ -1132,6 +1132,31 @@ public function sole($key = null, $operator = null, $value = null)
         return $items->first();
     }
 
+    /**
+     * Get the first item in the collection, but if no items exist throw an exception.
+     *
+     * @param  mixed  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
+     * @return mixed
+     *
+     * @throws \Illuminate\Collections\ItemNotFoundException
+     */
+    public function firstOrFail($key = null, $operator = null, $value = null)
+    {
+        $filter = func_num_args() > 1
+            ? $this->operatorForWhere(...func_get_args())
+            : $key;
+
+        $items = $this->when($filter)->filter($filter);
```

Comment:
In preparation for Laravel 9 (see [here](https://github.com/laravel/framework/pull/37667/commits/8eb3545c3ce8d0cf6b438844c9661b279f2a0691))

```suggestion
        $items = $this->unless($filter == null)->filter($filter);
```